database/sqlserver/sqlserver_test.go

```diff
@@ -218,3 +222,43 @@ func TestLockWorks(t *testing.T) {
 		}
 	})
 }
+
+func TestMsiTrue(t *testing.T) {
+	dktesting.ParallelTest(t, specs, func(t *testing.T, c dktest.ContainerInfo) {
+		ip, port, err := c.Port(defaultPort)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		addr := msConnectionStringMsi(ip, port, true)
+		p := &SQLServer{}
+		_, err = p.Open(addr)
+		if err == nil {
```

Comment:
Is there anyway to test the success case? e.g. are there mock providers available as docker images?

I'm happy to merge this in without the happy case tests but can't provide official support unless there are sufficient tests. e.g. w/o the happy case test, you'll need to mark this parameter as not officially supported in the README

Reply:
There doesn't seem to be a way to mock this through container images. The SQL Server container docs only go through password auth and don't mentioned MSI/AAD auth, which kinda makes sense since they're native to Azure, but would be nice if they provided mock images to test auth. We could mock MSI auth in code by returning an empty token, but that would only work if we were mocking the SQL Server instance as well and could control the auth on both sides. For now I will add a disclaimer in the README saying it's not officially supported.

Also, we have tested the happypath manually in our own scenarios and it does work.